typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
```

Comment:
Discussion for recommended rule: `'@typescript-eslint/no-unsafe-call': 'error'`

Reply:
Should we just enable [no-explicit-any](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-explicit-any.md) instead?

Reply:
Discussed in person: we think it's ok to have both no-explicit-any and this rule enabled so let's leave the rule configured as it is in the recommended config.

Reply:
As a group decided having the rule and `no-explicit-any` makes sense. A [separate PR](https://github.com/ni/javascript-styleguide/pull/30) to elevate the recommended `no-explicit-any` rule from `warn` to `error`.